### PR TITLE
99 housing recovery model analysis

### LINKED ISSUE #99: Housing recovery model analysis
The housing recovery model using Census appraisal data, Galveston testbed notebook example. 
Notes: https://opensource.ncsa.illinois.edu/confluence/display/INCORE1/Housing+recovery+analysis

### PR DESCRIPTION
**Pull request by ncsa-mo**
close #99

### CODE REVIEW

**Comment by longshuicy** [comment]
I'm getting an `Failed to download the data from Census API. Please check your parameters!` error

**Comment by longshuicy** [comment]
> I'm getting an `Failed to download the data from Census API. Please check your parameters!` error

Disregard this comment. It's working for me now.

**Review by longshuicy — APPROVED** [review]

File: pyincore/analyses/housingrecovery/housingrecovery.py [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+                   "id": "census_appraisal_data",
+                   "required": False,
+                   "description": "Census data, 2010 Decennial Census District (GCAD) Census data",
+                   "type": ["incore:censusAppraisalData"]
+                }
+            ],
+            "output_datasets": [
+                {
+                   "id": "result",
+                   "description": "A csv file with the building values for the 6 years following the disaster"
```
Comment (vnarah2) [inline pyincore/analyses/housingrecovery/housingrecovery.py:102]: can you update the description to say that it also returns what `hse_rec_fin_index` represents, so it's clear what the `index_n` columns mean in the outputs. 

Might also be helpful to say year -1 represents pre-impact conditions

**Review by vnarah2 — CHANGES_REQUESTED** [review]
This analysis requires pyincore-data, so should we be adding it as a dependency in requirements.txt, and also meta.yml? Tagging @ywkim312 for his inputs too

File: pyincore/analyses/housingrecovery/housingrecovery.py (hunk 1 above)
Comment (ncsa-mo) [reply pyincore/analyses/housingrecovery/housingrecovery.py:102]: Done

**Comment by ywkim312** [comment]
> 

Yes I think that we have to add pyincore-data>=0.4.0 should be there. But it might fail to build until we post 0.4.0 in conda channel or pypi

**Comment by navarroc** [comment]
Yeah, we probably do need to add the pyincore-data dependency to pyincore. I'm curious what the implication is on building the conda package (e.g. do we have to release pyincore-data first so we make sure the release gets the pyincore-data package it needs).

**Comment by ywkim312** [comment]
> Yeah, we probably do need to add the pyincore-data dependency to pyincore. I'm curious what the implication is on building the conda package (e.g. do we have to release pyincore-data first so we make sure the release gets the pyincore-data package it needs).

yes I believe so, it has to be published in the conda first. And we need to have pyincore-data>=0.4.0

**Comment by ywkim312** [comment]
> > Yeah, we probably do need to add the pyincore-data dependency to pyincore. I'm curious what the implication is on building the conda package (e.g. do we have to release pyincore-data first so we make sure the release gets the pyincore-data package it needs).
> 
> yes I believe so, it has to be published in the conda first. And we need to have pyincore-data>=0.4.0

If needed, I can publish current release branch as 0.4.0 version than update again after merge to main.

File: pyincore/analyses/housingrecovery/housingrecovery.py [hunk 2]
```diff
@@ -0,0 +1,395 @@
+# This program and the accompanying materials are made available under the
+# terms of the Mozilla Public License v2.0 which accompanies this distribution,
+# and is available at https://www.mozilla.org/en-US/MPL/2.0/
+
+import sys
+import numpy as np
+import pandas as pd
+
+from pyincore_data.censusutil import CensusUtil
```
Comment (navarroc) [inline pyincore/analyses/housingrecovery/housingrecovery.py:9]: Need to remove the import for Census

File: pyincore/analyses/housingrecovery/housingrecovery.py [hunk 3]
```diff
@@ -128,22 +122,15 @@ def run(self):
...
         bg_mhhinc = self.get_input_dataset("census_block_groups_data").get_dataframe_from_csv(low_memory=False)
 
-        # Census API using pyincore_data method
-        columns_api = "GEO_ID,NAME,B25002_001E,B25002_001M,B25004_006E,B25004_006M"
-        fips = self.get_parameter("fips")
-        if fips is None:
-            if self.get_input_dataset("census_appraisal_data") is None:
+        # Census data
+        if self.get_input_dataset("census_appraisal_data") is None:
                 sys.exit("Census data is required!.")
```
Comment (navarroc) [inline pyincore/analyses/housingrecovery/housingrecovery.py:125]: I don't think either of the sys.exits are needed. It's a required input so it should fail earlier with missing required input dataset.

**Review by navarroc — APPROVED** [review]

File: pyincore/analyses/housingrecovery/housingrecovery.py [hunk 4]
```diff
@@ -291,7 +273,7 @@ def merge_seasonal_data(self, hse_rec, vac_status):
         vac_status["tractid"] = vac_status["state"].astype(str) + vac_status["county"].astype(str) + \
                                 vac_status["tract"].astype(str)
 
-        hse_rec_merged = pd.merge(hse_rec, vac_status, on="tractid", how="inner")
+        hse_rec_merged = pd.merge(hse_rec, vac_status, left_on="tractid", right_on="tractid", how='inner')
```
Comment (vnarah2) [inline pyincore/analyses/housingrecovery/housingrecovery.py:276]: aren't these two lines equivalent? left+right vs on for inner merge.

Just wondering if this was done to fix something

**Review by vnarah2 — APPROVED** [review]

File: pyincore/analyses/housingrecovery/housingrecovery.py (hunk 4 above)
Comment (ncsa-mo) [reply pyincore/analyses/housingrecovery/housingrecovery.py:276]: It is the same thing. It probably stayed from data validation. There was a test dataset with tract (not tractid) on the "right" side.

**Pull request merged**